scripts/nlp_language_modeling/build_retrieval_index.py

```diff
@@ -34,11 +35,67 @@
 It creates a index.sav which can be loaded by Faiss. It can look up the KNN chunk ids of the 
 DB dataset given the input embedding vector. 
 
+To use it in multiple stages, it follows the example as shown in  
+https://github.com/facebookresearch/faiss/blob/main/demos/demo_ondisk_ivf.py
+
+stage-0: train on the dataset, example,
+
+```python
+python scripts/nlp_language_modeling/build_retrieval_index.py \
```

Comment:
Training uses the entirety of the dataset right?  Can you tell a bit what `train_index_size` & `train_chunk_size` do here?